src/auth/ui/vue/show.ts

```diff
@@ -332,18 +351,382 @@ export class AuthWebview extends VueWebview {
+            credentialSourceId: args.authType,
+            featureId: args.featureType,
+            result: args.reason === userCancelled ? 'Cancelled' : 'Failed',
+            reason: args.reason,
+            invalidInputFields: args.invalidInputFields
+                ? builderCommaDelimitedString(args.invalidInputFields)
+                : undefined,
+            isAggregated: false,
+        })
+
+        if (
+            this.#previousFailedAuth &&
+            this.#previousFailedAuth.authType === args.authType &&
+            this.#previousFailedAuth.featureType === args.featureType
+        ) {
+            // Another failed attempt on same auth + feature. Update with newest failure info.
+            this.#previousFailedAuth = {
+                ...this.#previousFailedAuth,
+                ...args,
+                attempts: this.#previousFailedAuth.attempts + 1,
+            }
+        } else {
+            // A new failed attempt on a new auth + feature
+            this.#previousFailedAuth = {
+                invalidInputFields: undefined, // args may not have this field, so we set the default
+                ...args,
+                attempts: 1,
+            }
+        }
+    }
+
+    /** Returns the number of failed attempts for the given auth form */
+    getPreviousAuthAttempts(featureType: FeatureId, authType: CredentialSourceId) {
+        if (
+            this.#previousFailedAuth &&
+            this.#previousFailedAuth.featureType === featureType &&
+            this.#previousFailedAuth.authType === authType
+        ) {
+            return this.#previousFailedAuth.attempts
+        }
+        return 0
+    }
+
+    async successfulAuthAttempt(args: { authType: CredentialSourceId; featureType: FeatureId }) {
+        // All previous failed attempts + 1 successful attempt
+        const authAttempts = this.getPreviousAuthAttempts(args.featureType, args.authType) + 1
+        this.emitAuthAttempt({
+            authType: args.authType,
+            featureType: args.featureType,
+            result: 'Succeeded',
+            attempts: authAttempts,
+        })
+    }
+
+    #totalAuthAttempts: number = 0
+
+    /**
+     * This metric is emitted on an attempt to signin/connect/submit auth regardless
+     * of success.
+     */
+    private async emitAuthAttempt(args: {
+        authType: CredentialSourceId
+        featureType: FeatureId
+        result: Result
+        reason?: string
+        invalidFields?: string[]
+        attempts: number
+    }) {
+        telemetry.auth_addConnection.emit({
+            source: this.#authSource ?? '',
+            credentialSourceId: args.authType,
+            featureId: args.featureType,
+            result: args.result,
+            reason: args.reason,
+            invalidInputFields: args.invalidFields ? builderCommaDelimitedString(args.invalidFields) : undefined,
+            attempts: args.attempts,
+            isAggregated: true,
```

Comment:
I'm a bit confused about this field. If the event is emitted on every attempt, how could it be aggregated? I thought `auth_addedConnections` was the aggregate event.

Reply:
`auth_addConnection` will have 2 different types essentially. If `isAggregated` is `true` it has information regarding the # of attempts prior to it and is the final result.

I did it this way to we can both emit a metric for every single interaction (isAggregated === false). Additionally, a final one (isAggregated === true) which is for cases like the user failed multiple times, but their final one was a success.

Once we want to know their final result of all attempts, we only need to look at the ones with `isAggregated === true`